social_engagement/engagement.py

```diff
@@ -262,36 +287,79 @@ def get_involved_users_in_thread(request, thread):
     """
     Compute all the users involved in the children of a specific thread.
     """
-    users = set()
     params = {"thread_id": thread.id, "page_size": 100}
-    is_question = True if getattr(thread, "thread_type", None) == "question" else False
+    is_question = getattr(thread, "thread_type", None) == "question"
+    author_id = getattr(thread, 'username', None)
+    if hasattr(thread, 'username'):
+        author_id = thread.user_id
+
+    results = _detail_results_factory()
+
     if is_question:
         # get users of the non-endorsed comments in thread
         params.update({"endorsed": False})
-        users.update(_get_users_in_thread(_get_request(request, params)))
+        _get_details_for_deletion(_get_request(request, params), results, is_thread=True)
         # get users of the endorsed comments in thread
         params.update({"endorsed": True})
-        users.update(_get_users_in_thread(_get_request(request, params)))
+        _get_details_for_deletion(_get_request(request, params), results, is_thread=True)
     else:
-        users.update(_get_users_in_thread(_get_request(request, params)))
+        _get_details_for_deletion(_get_request(request, params), results, is_thread=True)
+
+    users = _extract_users_from_results(results)
+
+    if author_id:
+        users[author_id]['num_upvotes'] += thread.votes.get('count', 0)
+        users[author_id]['num_threads'] += 1
+        users[author_id]['num_comments_generated'] += results['all_comments']
+        users[author_id]['num_thread_followers'] += thread.get_num_followers()
+
     return users
 
 
 def get_involved_users_in_comment(request, comment):
-    '''
+    """
     Method used to extract the involved users in the comment.
     This method also returns the creator of the post.
-    '''
-    users = set()
-    users.update(_get_users_in_comment(request, comment.id))
+    """
+    params = {"page_size": 100}
+    author_id = None
     if hasattr(comment, 'parent_id'):
-        users.add(_get_author_of_comment(comment.parent_id))
+        author_id = _get_author_of_comment(comment.parent_id)
     if hasattr(comment, 'thread_id'):
-        users.add(_get_author_of_thread(comment.thread_id))
+        author_id = _get_author_of_thread(comment.thread_id)
+
+    results = _get_details_for_deletion(_get_request(request, params), comment.id)
+    users = _extract_users_from_results(results)
+
+    if author_id:
+        users[int(author_id)]['num_replies'] += results['replies']
+
+    return users
+
 
+def _extract_users_from_results(results):
+    """
+    Helper method for getting involved users from results.
+    """
+    users = results['users']
+    for user in users.keys():
+        id_ = int(User.objects.get(username=user).id)
```

Comment:
> Nit: usually private vars/functions will prefix the underscore rather than postfix. Either way, do we need to put any underscore? I don't see any other `id` being used within this function that could conflict with this.

Actually, [`id()`](https://docs.python.org/2/library/functions.html#id) is a built-in in Python, so I've named this variable according to PEP-8's [Descriptive: Naming Styles](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles):

> single_trailing_underscore_: used by convention to avoid conflicts with Python keyword, e.g.
> `https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles`

Reply:
> this seems to be making a database call for every single user in the dictionary! We need to make a bulk call and then iterate over the result of that instead. So basically filter with `username__in=users.keys()` first and then iterate over that queryset.

Ah, good point!

Reply:
Ah alright, didn't know about that particular styling point, but sounds good.